src/System.Linq.Expressions/src/System/Runtime/CompilerServices/ReadOnlyCollectionBuilder.cs

```diff
@@ -131,7 +128,7 @@ public int IndexOf(T item)
         /// <param name="item">The object to insert into the <see cref="ReadOnlyCollectionBuilder{T}"/>.</param>
         public void Insert(int index, T item)
         {
-            if (index > _size)
+            if (index < 0 || index > _size)
```

Comment:
This appears to be a behavioral change, in that a different exception could result than before.  Is that the case?